### PR TITLE
docs(en-US): references -> users -> session and cookie

### PR DESCRIPTION
**Pull request by IceHe**
References -> Users → Session and cookie

### CODE REVIEW

File: docs/docs/references/users/README.md [hunk 1]
```diff
@@ -261,3 +261,54 @@ Every time the user signs in with a social connector,
 their _identities_ will be automatically imported or updated from the identity provider.
 
 :::
+
+## Session and cookie
+
+A session is a group of interactions between a user and an application that takes place (maybe on different pages) within a period of time.
+A single session temporarily stores multiple interactions (such as sign-in, linking to a social network, consent authorization, and sign-out) before it ends.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:268]: social network?

File: docs/docs/references/users/README.md [hunk 2]
```diff
@@ -261,3 +261,54 @@ Every time the user signs in with a social connector,
...
+If the user leaves a website or closes the browser, the session will end by default.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:269]: i don;t think we need this.. our reader has the basic concept of session or cookie or they can just go wikipedia. we just need to talk about how session and cookie works in logto.

File: docs/docs/references/users/README.md (hunk 1 above)
Comment (IceHe) [inline docs/docs/references/users/README.md:268]: Updated to
```suggestion
A single session temporarily stores multiple interactions (such as sign-in, linking to a social identity provider, consent authorization, and sign-out) before it ends.
```

File: docs/docs/references/users/README.md (hunk 2 above)
Comment (IceHe) [reply docs/docs/references/users/README.md:269]: Can I add a link for the freshman to read?

Comment (IceHe) [reply docs/docs/references/users/README.md]: reorg

File: docs/docs/references/users/README.md [hunk 3]
```diff
@@ -261,3 +261,49 @@ Every time the user signs in with a social connector,
 their _identities_ will be automatically imported or updated from the identity provider.
 
 :::
+
+## Session and cookie
+
+It's inconvenient to sign in every time the user re-visits the application.
+The application using Logto can extend the session by putting its information in
+[cookies](https://en.wikipedia.org/wiki/HTTP_cookie) for reuse to avoid this inconvenience.
+The extended session will end when it expires, the user signs out on purpose, or the password has been changed.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:270]: how do we end session when password changed?

File: docs/docs/references/users/README.md [hunk 4]
```diff
@@ -261,3 +261,49 @@ Every time the user signs in with a social connector,
...
+([Time To Live](https://en.wikipedia.org/wiki/Time_to_live)).
+After the session expires (i.e., its _lifetime_ finishes), the user must sign in again.
+
+### Session layers
+
+Using Logto as an identity provider, three layers of sessions may be created:
+
+- **Application Layer**: When a user signs in using Logto and goes to your application, you need to keep track of the authentication info, such as the "authenticated" state.
+  Keeping track of them usually means saving a set of tokens in the storage provided by your application.
+  The storage can be cookies in a regular web application.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:285]: this is unclear. do we use SDK to archive this? if so we are not using cookies

File: docs/docs/references/users/README.md [hunk 5]
```diff
@@ -261,3 +261,49 @@ Every time the user signs in with a social connector,
...
+- **Logto Layer**: Logto keeps track of the session through the browser's cookies.
+  After the session expires, the user may be redirected to Logto sign-in page and need to enter the authentication info again.
+- **Social Sign-in Layer**: When the user is redirected to Logto sign-in page and tries to sign in with a social connector, an additional redirection to the social identity provider is needed.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:288]: we don't care about social sign-in here. you were talking about "Logto layer session" in the previous section but talks a broader "session" here. very confusing

File: docs/docs/references/users/README.md [hunk 6]
```diff
@@ -261,3 +261,49 @@ Every time the user signs in with a social connector,
 their _identities_ will be automatically imported or updated from the identity provider.
 
 :::
+
+## Session and cookie
+
+It's inconvenient to sign in every time the user re-visits the application.
+The application using Logto can extend the session by putting its information in
+[cookies](https://en.wikipedia.org/wiki/HTTP_cookie) for reuse to avoid this inconvenience.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:269]: and i still don't know what kind of session and cookie that logto holds

Comment (gao-sun) [reply docs/docs/references/users/README.md:269]: and you also missed the cookie signing part

File: docs/docs/references/users/README.md [hunk 7]
```diff
@@ -261,3 +261,49 @@ Every time the user signs in with a social connector,
...
+
+### Configuration
+
+You can customize the session's behaviors by changing the [oidc-provider](https://github.com/panva/node-oidc-provider)
+configuration in the Logto Core [code](https://github.com/logto-io/logto/blob/master/packages/core/src/oidc/init.ts).
+See the oidc-provider's [documentation](https://github.com/panva/node-oidc-provider/blob/main/docs/README.md) for details.
+
+## Manage users
+
+You can check out [Manage users](../../recipes/manage-users/) for details.
```
Comment (simeng-li) [inline docs/docs/references/users/README.md:309]: If everything is under the recipes, should we list the Manage Users here? Is it part of the user's reference? Could just be a info note?

**Comment by IceHe** [comment]
Per Slack thread [discussion](https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1656673305620939).
LOG-2651 would be better taken with another teammate that is more familiar with this part.

**Pull request closed without merging**